node/sc/vt_recovery.go

```diff
@@ -252,19 +256,42 @@ func retrievePendingEventsFrom(hint *valueTransferHint, br *bridge.Bridge) ([]*R
 	}
 
 	var pendingEvents []*RequestValueTransferEvent
-	it, err := br.FilterRequestValueTransfer(&bind.FilterOpts{Start: hint.blockNumber}) // to the current
+
+	curBlkNum, err := bi.GetCurrentBlockNumber()
 	if err != nil {
 		return nil, err
 	}
-	for it.Next() {
-		logger.Trace("pending nonce in the event", "requestNonce", it.Event.RequestNonce)
-		if it.Event.RequestNonce >= hint.handleNonce {
-			logger.Trace("filtered pending nonce", "requestNonce", it.Event.RequestNonce, "handledNonce", hint.handleNonce)
-			pendingEvents = append(pendingEvents, &RequestValueTransferEvent{it.Event})
+
+	done := false
+	startBlkNum := hint.blockNumber
+	endBlkNum := startBlkNum + filterLogsStride
+
+pendingTxLoop:
+	for !done && startBlkNum <= curBlkNum {
+		if endBlkNum > curBlkNum {
+			endBlkNum = curBlkNum
+			done = true
 		}
+		opts := &bind.FilterOpts{Start: startBlkNum, End: &endBlkNum}
+		it, err := bi.bridge.FilterRequestValueTransfer(opts)
+		if err != nil {
+			return nil, err
+		}
+		for it.Next() {
+			logger.Trace("pending nonce in the event", "requestNonce", it.Event.RequestNonce)
+			if it.Event.RequestNonce >= hint.handleNonce {
+				logger.Trace("filtered pending nonce", "requestNonce", it.Event.RequestNonce, "handledNonce", hint.handleNonce)
+				pendingEvents = append(pendingEvents, &RequestValueTransferEvent{it.Event})
+				if len(pendingEvents) > maxPendingTxs {
```

Comment:
How about checking like below?
I think maxPendingTxs means this can process maxmimum `maxPendingTxs` number of events.
```suggestion
				if len(pendingEvents) >= maxPendingTxs {
```

Reply:
Fixed.